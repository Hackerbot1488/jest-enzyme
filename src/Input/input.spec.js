import React from 'react'
import Input from '../Select/select';

describe('Input component', () => {
  it('should render input component', () => {
    const component = shallow(<Input />)
    expect(component).toMatchSnapshot()
  });
  /* it('should call onChange method', () => {
    const mockCallBAck = jest.fn()
    const component = shallow(<Input onChange={mockCallBAck} />)
    expect(mockCallBAck.mock.calls.length).toBe(0)
    component.find("input").simulate("change")
    expect(mockCallBAck.mock.calls.length).toEqual(1)
  }); */
  describe('defaultProps', () => {
    it('should use default onChange', () => {
      const result = Input.defaultProps.onChange;
      expect(result).toBe(undefined)
    });
    it('should use default onKeyPress', () => {
      const result = Input.defaultProps.onKeyPress;
      expect(result).toBe(undefined)
    });
  });
});
