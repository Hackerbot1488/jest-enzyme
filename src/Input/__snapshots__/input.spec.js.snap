// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render input component 1`] = `
<div
  className="selectWrapper"
>
  <div
    className="placeholder"
  >
    "No items"
  </div>
</div>
`;
